main.rb

```diff
@@ -0,0 +1,163 @@
+require './book'
+require './person'
+require './student'
+require './rental'
+require './teacher'
+
+@books_arr = []
+@people_arr = []
+@rentals_arr = []
+
+def book_store(book)
+  @books_arr.push(book)
+  @books_arr
+end
+
+def people_store(person)
+  @people_arr.push(person)
+  @people_arr
+end
+
+def rentals_store(rental)
+  @rentals_arr.push(rental)
+  @rentals_arr
+end
+
+def success(message)
+  puts "Successfully created #{message}"
+end
+
+def user_info
+  print 'Enter the name of the person: '
+  name = gets.chomp
+  print 'Age: '
+  age = gets.chomp
+  [name, age]
+end
+
+def create_student
+  name, age = user_info
+  print 'Has parent permission? [Y/N]: '
+  parent_permission = gets.chomp
+  print 'Enter the classroom: '
+  classroom = gets.chomp
+  case parent_permission.upcase
+  when 'Y'
+    permission = true
+  when 'N'
+    permission = false
+  else
+    puts 'Invalid input'
+    run
+  end
+  student = Student.new(classroom, age, name, parent_permission: permission)
+  people_store(student)
+  success('Student')
+end
+
+def create_teacher
+  # destructure name and age from user info
+  name, age = user_info
+  puts 'Enter the specialization: '
+  specialization = gets.chomp
+  teacher = Teacher.new(specialization, age, name)
+  people_store(teacher)
+  success('Teacher')
+end
+
+# Menu Operations
+
+def list_books
+  @books_arr.each do |book|
+    puts "#{book.title} by #{book.author}"
+  end
+  run
+end
+
+def list_people
+  @people_arr.each { |person| puts "#{person.name} - #{person.age}" }
+  run
+end
```

Comment:
- [ ] It will be better if we can differentiate between the teacher and the student when listing people. What did you think?. This can easily be done by printing the class to which each person belongs ie `person.class`. It will be nice if we can also include the ID of each person as seen in the provided [video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) 
 

- **Original**
![Screenshot from 2022-03-15 20-48-47](https://user-images.githubusercontent.com/63087491/158463067-b4b44c45-e649-4c42-bcd6-e1c56879fdb0.png)

- **Yours**
![Screenshot from 2022-03-15 20-49-12](https://user-images.githubusercontent.com/63087491/158463095-abc4b497-7e34-4716-81d7-a75fe990cba2.png)